lib/transmuxer/mp4_output_transmuxer.js

```diff
@@ -0,0 +1,1184 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.provide('shaka.transmuxer.Mp4OutputTransmuxer');
+
+goog.require('shaka.transmuxer.ADTS');
+goog.require('shaka.transmuxer.Ac3');
+goog.require('shaka.transmuxer.Ec3');
+goog.require('shaka.transmuxer.H264');
+goog.require('shaka.transmuxer.H265');
+goog.require('shaka.transmuxer.MpegAudio');
+goog.require('shaka.util.BufferUtils');
+goog.require('shaka.util.Error');
+goog.require('shaka.util.Id3Utils');
+goog.require('shaka.util.IReleasable');
+goog.require('shaka.util.ManifestParserUtils');
+goog.require('shaka.util.MimeUtils');
+goog.require('shaka.util.Mp4Generator');
+goog.require('shaka.util.TsParser');
+goog.require('shaka.util.Uint8ArrayUtils');
+
+goog.requireType('shaka.media.SegmentReference');
+
+
+/**
+ * MP4 output transmuxer
+ *
+ * @implements {shaka.util.IReleasable}
+ */
+shaka.transmuxer.Mp4OutputTransmuxer = class {
```

Comment:
I'm not convinced that this design with everything in one class is right, either.

 - The individual Transmuxer classes implementing shaka.extern.Transmuxer seem pointless now.
 - This fix will be nearly impossible to cherry-pick.

Could you maybe have format probing driven at a higher level, when we're selecting which shaka.extern.Transmuxer to invoke?  You could add a probe function to the interface and use it to select the right one if audio should be TS but isn't.

Reply:
We need detect it in each segment, see https://github.com/shaka-project/shaka-player/issues/6357 (the same media playlist has aac (ads) and TS (main content)).

I know that the fix is impossible to cherry-pick…

Tomorrow I will think of some other solution, but I don't find many valid ones...